src/sw.js

```diff
@@ -16,40 +16,39 @@
  * along with this program. If not, see http://www.gnu.org/licenses/.
  *
  */
+/* eslint-disable */
 
-(global => {
-  global.importScripts('/worker/sw-toolbox.js');
-  global.importScripts('/worker/lru-cache-strategy.js');
+import {skipWaiting, clientsClaim} from 'workbox-core';
+import {registerRoute} from 'workbox-routing';
+import {CacheFirst} from 'workbox-strategies';
+import {ExpirationPlugin} from 'workbox-expiration';
 
-  const ASSET_CACHE_MAX_ITEMS = 1000;
-  const CACHE_VERSION = 2;
-  const CURRENT_CACHES = {
-    asset: `asset: 'asset-cache-v${CACHE_VERSION}`,
-  };
+skipWaiting();
+clientsClaim();
 
-  global.toolbox.options.debug = false;
-  global.toolbox.router.default = global.toolbox.networkOnly;
-  global.toolbox.router.get(/forceCaching=true/, global.cacheLRU, {
-    cache: {
-      maxEntries: ASSET_CACHE_MAX_ITEMS,
-      name: CURRENT_CACHES.asset,
-    },
-  });
+self.__WB_MANIFEST;
 
-  global.addEventListener('install', event => event.waitUntil(global.skipWaiting()));
-  global.addEventListener('activate', event => {
-    const expectedCacheNames = Object.values(CURRENT_CACHES);
+const ASSET_CACHE_MAX_ITEMS = 1000;
+const CACHE_VERSION = 3;
+const ASSET_CACHE_NAME = `asset-cache-v${CACHE_VERSION}`;
 
-    return event.waitUntil(
-      global.caches.keys().then(cacheNames => {
-        return Promise.all(
-          cacheNames.map(cacheName => {
-            if (!expectedCacheNames.includes(cacheName)) {
-              return global.caches.delete(cacheName);
-            }
-          }),
-        ).then(() => global.clients.claim());
+registerRoute(
+  ({url}) => {
+    return url.search.includes('forceCaching=true');
+  },
+  new CacheFirst({
+    cacheName: ASSET_CACHE_NAME,
+    cacheableResponse: {
+      statuses: [200],
+    },
+    plugins: [
+      new ExpirationPlugin({
+        maxAgeSeconds: 7 * 24 * 60 * 60, // One week
```

Comment:
Is one week okay? Previously it was not limited because of the effort to implement that.

Reply:
Yepp. Let's go with one week. 👍